src/utils/users.js

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import Client from 'client';
+import {UsersTypes} from 'constants';
+const HTTP_UNAUTHORIZED = 401;
+
+export function profilesToSet(state, action) {
+    const nextState = {...state};
+    if (action.offset != null && action.count != null) {
+        nextState.offset = action.offset + action.count;
+        nextState.count += action.count;
+    }
+    Object.keys(action.data).forEach((key) => {
+        if (!nextState.items.has(key)) {
+            nextState.items.add(key);
```

Comment:
This is definitely mutating the state. If key here is a string, it'd be fine to use JS objects as pseudo sets, but if they're more complicated objects, we'll need to bring in ImmutableJS or roll our own.